User.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace common;
+
+use yii\db\ActiveRecord;
+
+class User extends ActiveRecord
+{
+    public static function getCommentsForPosts()
+    {
+        $posts = Post::find()->where(['user_id' => $this->id])->all();
+        $result = [];
```

Comment:
$result is no more needed
```suggestion
 $comments = Comment::model()->findAllByAttributes(array('post_id' => $posts));
```